--- ModuleScopePlugin.js	2021-01-11 13:06:00.000000000 -0800
+++ ModuleScopePlugin-patched.js	2021-01-11 13:06:10.000000000 -0800
@@ -30,6 +30,11 @@
           // If this resolves to a node_module, we don't care what happens next
           request.descriptionFileRoot.indexOf('/node_modules/') !== -1 ||
           request.descriptionFileRoot.indexOf('\\node_modules\\') !== -1 ||
+          // --- BEGIN StackRox Patch ---
+          // Due to an issue in `react-scripts` CSS imports don't work properly with yarn workspaces
+          // as symlinks in `node_modules` get resolved too early.
+          (request.descriptionFileRoot.indexOf('/ui/packages/') !== -1 && request.path.endsWith('.css')) ||
+          // --- END StackRox Patch ---
           // Make sure this request was manual
           !request.__innerRequest_request
         ) {
